python/src/aiconfig/editor/client/src/components/aiconfigReducer.ts

```diff
@@ -134,6 +141,20 @@ export default function aiconfigReducer(
         name: action.name,
       }));
     }
+    case "UPDATE_PROMPT_MODEL": {
+      return reduceReplacePrompt(state, action.index, (prompt) => ({
+        ...prompt,
+        metadata: {
+          ...prompt.metadata,
+          model: action.modelName
+            ? {
+                name: action.modelName,
+                // TODO: Consolidate settings based on schema union
+              }
+            : undefined,
```

Comment:
Hopefully don't need to do this logic with updated SDK

Reply:
I'll revisit this once the endpoint is implemented. We still need some UI state change here (this happens before the request is sent and before the response is returned to ensure responsive UI). Probably what will need to be done is:
- this action sets just the name
- aiconfig response comes back with full metadata, either cleared settings or updated settings to match supported settings based on the schema
- from aiconfig response, we consolidate the UI state model metadata with updated settings